src/federation.rs

```diff
@@ -0,0 +1,268 @@
+use crate::crypto::multi_party_schnorr::SharedKeys;
+use crate::crypto::vss::Vss;
+use crate::errors::Error;
+use crate::net::SignerID;
+use crate::sign::Sign;
+use crate::signer_node::{SharedSecret, SharedSecretMap};
+use bitcoin::PublicKey;
+use curv::cryptographic_primitives::secret_sharing::feldman_vss::{
+    ShamirSecretSharing, VerifiableSS,
+};
+
+#[derive(Debug)]
+pub struct Federations {
+    /// The vector of federations. This vector should be sorted by block height.
+    vec: Vec<Federation>,
+}
+
+impl Federations {
+    pub fn new(federations: Vec<Federation>) -> Self {
+        let mut federations = federations;
+        federations.sort_by_key(|f| f.block_height());
+        Federations { vec: federations }
+    }
+
+    pub fn len(&self) -> usize {
+        self.vec.len()
+    }
+
+    pub fn validate(&self) -> Result<(), Error> {
+        if self.vec.len() == 0 {
```

Comment:
Should we allow two federations which have same block_height?

Reply:
I think It should be avoided. I will add a check for it.

Reply:
Fixed at ec71ff3192e4a76cc062e64ae85fb5b261bf9967